plugins/modules/iptables_state.py

```diff
@@ -260,42 +257,48 @@ def read_state(b_path):
     return [t for t in text.splitlines() if t != ""]
 
 
-def write_state(b_path, lines, changed):
+def get_file_contents(b_path: bytes) -> bytes | None:
+    try:
+        with open(b_path, "rb") as f:
+            return f.read()
+    except FileNotFoundError:
+        return None
+
+
+def write_state(module: AnsibleModule, b_path: bytes, lines: list[str], changed: bool) -> bool:
     """
     Write given contents to the given path, and return changed status.
     """
-    # Populate a temporary file
-    tmpfd, tmpfile = tempfile.mkstemp()
-    with os.fdopen(tmpfd, "w") as f:
-        joined_lines = "\n".join(lines)
-        f.write(f"{joined_lines}\n")
-
-    # Prepare to copy temporary file to the final destination
-    if not os.path.exists(b_path):
+    joined_lines = "\n".join(lines)
+    content = f"{joined_lines}\n".encode("utf-8")
```

Comment:
I know the end result is going to be the exact same, just trying to establish rules and criteria in my mind: when should we use `.encode()` like that, and when should we use `to_bytes()` from core utils? I know this was more important when Python 2 was in the equation, but is it any different now? Should we adopt one idiom over the other, or just let it go either way?